pkg/operator/controller/status_reconciler/status_reconciler.go

```diff
@@ -48,7 +48,17 @@ func (r *StatusReconciler) ReconcileStatus(ctx context.Context, object objectWit
 	if err := r.reconcileCondition(ctx, object, r.getAvailableCondition(ctx, object.GetGeneration())); err != nil {
 		return err
 	}
-	return r.reconcileCondition(ctx, object, r.getDependenciesFulfilledCondition(object.GetGeneration()))
+	if err := r.reconcileCondition(ctx, object, r.getDependenciesFulfilledCondition(object.GetGeneration())); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *StatusReconciler) ReconcileStatusWithConfig(ctx context.Context, object objectWithConditions, kaiConfig *kaiv1.Config) error {
+	if err := r.ReconcileStatus(ctx, object); err != nil {
+		return err
+	}
+	return r.reconcileCondition(ctx, object, r.getPrometheusConnectivityCondition(ctx, object.GetGeneration(), kaiConfig))
```

Comment:
Lets find a way to call this from the operand's code. If you need to you can add to the interface.